src/test/futex/test_futex.c

```diff
@@ -203,7 +203,7 @@ static void _futex_wait_bitset_test() {
     }
 
     // Wait a bit until they're (hopefully) all blocked on the futex.
-    usleep(1);
+    usleep(10000);
```

Comment:
Hmm, this is an ok workaround for now, but ideally we should avoid there being a race condition here instead of just making it less likely.

In the current code the assertions on 236-239 are unsafe, since the thread might not have gone to sleep before the futex was released.

Minimally, please add a comment on 235 about the race condition. Maybe comment out 236-239 until it can be fixed, though we could wait and see if the longer sleep here is good enough in practice.

I think the only way to make those assertions safely might be to sleep in a loop until we successfully validate that the threads are blocked; maybe by checking field 3 of `/proc/<tid>/stat`?